.github/workflows/conformance-k8s-kind-network-policies.yaml

```diff
@@ -39,11 +39,12 @@ jobs:
       IP_FAMILY: ${{ matrix.ipFamily }}
 
     steps:
-      - name: Checkout main branch to access local actions
+      - name: Checkout target branch to access local actions
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
         with:
-          ref: ${{ github.event.repository.default_branch }}
+          ref: ${{ github.base_ref || github.ref }}
```

Comment:
How do we protect against malicious PRs?

Reply:
> It seems to me like the intent here is for it to be a fallback for when github.base_ref is unavailable, maybe for scheduled runs on a stable branch(?)

Exactly, the idea being that for `pull_requests` events we check out the target branch (i.e., main in this case), and the branch itself (main again) for push events. According to the docs, `base_ref` is only available for `pull_request` events:

>  The base_ref or target branch of the pull request in a workflow run. This property is only available when the event that triggers a workflow run is either pull_request or pull_request_target.

Overall, this approach aims at producing exactly the same behavior as before (i.e., always checking out the main branch) for PRs and push events against main, while  checking out the specific stable branch when operating against stable branches. Another approach could be replacing `${{ github.event.repository.default_branch }}` with the given stable branch name on stable branches, but that can get out of control pretty soon. The `||` approach is also consistent with how we handle the [same operation in workflows triggered by Ariane](https://github.com/cilium/cilium/blob/f92b528abc30a5c66ba80d5922d4e58c48cfe7e1/.github/workflows/conformance-clustermesh.yaml#L192) (I've just realized that there we use `github.sha`, while here `github.ref` -- I've kept `ref` for consistency with `base_ref`, but happy to change if you have preferences).

That said, to my understanding, it wouldn't matter from a security PoW even if we directly checked out the PR branch in these workflows, as they are not running in a trusted context (i.e., `pull_request_target`) and PRs from forks are prevented by design the access to secrets. Additionally `pull_request` events leverage the workflow file from the PR branch itself, which means that they can be modified at will by the author of the PR (I couldn't find any quote from the official doc, but [here it is the proof](https://github.com/cilium/cilium/actions/runs/8184777228/job/22379856762?pr=31219#step:2:1)). However, the only consequence could be increasing the duration of the workflow on purpose, which could be done as well just modifying Cilium itself. All in all, we could just drop that additional step altogether, and always checkout `github.sha` only. I preferred preserving it, though, for consistency with the other workflows, and to prevent surprises if we ever decided to change how which events can trigger the workflow (e.g., switching to `pull_reqquest_target` for whatever reason).